source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -364,6 +374,54 @@ The annotations in the example POJO specify the following behavior:
 - Omit the ``relatedItems`` field and value when converting data
 - Use the ``Product(String name)`` constructor when instantiating the POJO
 
+
+.. _bsonextraelements-annotation-code-example:
+
+BsonExtraElements Example
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``@BsonExtraElements`` annotation allows you to specify a field to
+deserialize data from a MongoDB document that lacks a corresponding POJO
+field mapping. Consider a situation in which your application needs to handle
+data that conforms to different data schema versions. When your POJO fields
+change to accommodate the schema, you can continue to read, but not
+update, the remainder of the MongoDB document data from the field that uses
```

Comment:
I'm not sure what you mean that you can read but not update the remainder of the data.  It seems like you can both read _and_ update.

Reply:
To clarify, you're referring to generally being able to update the data and not when using a POJO field with the BsonExtraElements annotation? When reading the [description of the annotation](https://jira.mongodb.org/browse/DOCSP-22048), I inferred that the BsonExtraElements annotation would decode to a POJO, but not encode the field's value and pass it back to MongoDB.

I think while rewriting this section, I placed this info in a spot that made it confusing as to whether it was a requirement of the hypothetical situation or general capabilities of using a POJO to work with MongoDB data.

Reply:
From reading your next comment, it sounds like you can update the unstructured part and that the field data gets encoded in an update operation. Will test it out and update the descriptions accordingly.

Reply:
Wasn't able to update the MongoDB data using a POJO field marked with the annotation, so I'm assuming you meant you could update POJO fields generally, but not with the BsonExtraElements annotation.

Reply:
I just tried it and it worked as I expected.  Can you share your reproducer?

Reply:
https://gist.github.com/ccho-mongodb/98a7a6839c2b7627fb0cc2637151b2fa

I was able to set the "material" field on the "metadata" subdocument (annotated with @BsonExtraElements) using a POJO only on insert. Any attempt to update that field using the POJO (shown in the gist) resulted in no changes.

Isn't that the expected behavior? If not, what am I doing incorrectly?

Reply:
OK, I think I see the problem.  The `metadata` field does not result in a sub-document.  The name of a field annotated with `BsonExtraElements` is actually ignored, and the fields within the document itself are added at the same level as the other fields.  So when you insert the `NewProductPojo` instance in your example, you should end up with:

```
{ 
    _id: "MDB0123",
   material: "iron"
}
```

and not

```
{ 
    _id: "MDB0123",
   metadata: { material: "iron"}
}
```

Does that clear it up?

Reply:
Yes, from your explanation, I think my original understanding matched what you're saying, but needed to be more specific in the wording of the explanation in the docs.

To summarize, you can change the underlying data, but not from the BsonExtraElements-annotated field since it's only decoded and not encoded. E.g. I can update the MongoDB document by directly updating the "material" field, but not from the "metadata" POJO field.

Reply:
I don't think what you're saying is correct.  It _is_ encoded, and either insertOne or replaceOne will write the context of the field annotated with BsonExtraElements into the top-level document.  

We can touch base tomorrow to clarify